### ISSUE TITLE
Sabnzbd completes, but Sonarr import fails for file in use (v4.0.0)

### ISSUE THREAD

**Issue by PearsonFlyer** [sabnzbd/sabnzbd#2470]
Occasionally when sab completes the download (status: Complete) since updating to 4.0.0 beta, it doesn't seem Sab completely lets go of the file.

Sonarr reports:

`Couldn't import episode I:\sab\complete\tv\The.Nevers.S01E09.1080p.WEBRip.AAC2.H264-NTb\The.Nevers.S01E09.1080p.WEBRip.AAC2.H.264-NTb.mkv: The process cannot access the file 'I:\Media\TV\Nevers, The (2021) [imdbid-tt8697870]\Season 01\Nevers, The (2021) S01E09 Fever [WEBRip-1080p]-BTN.mkv' because it is being used by another process.`

And these additional details in Sonarr's logs:

```System.IO.IOException: The process cannot access the file 'I:\Media\TV\Nevers, The (2021) [imdbid-tt8697870]\Season 01\Nevers, The (2021) S01E09 Fever [WEBRip-1080p]-BTN.mkv' because it is being used by another process.

   at System.IO.FileSystem.DeleteFile(String fullPath)

   at NzbDrone.Core.MediaFiles.RecycleBinProvider.DeleteFile(String path, String subfolder) in C:\BuildAgent\work\13f3e374fa512e16\src\NzbDrone.Core\MediaFiles\RecycleBinProvider.cs:line 84

   at NzbDrone.Core.MediaFiles.UpgradeMediaFileService.UpgradeEpisodeFile(EpisodeFile episodeFile, LocalEpisode localEpisode, Boolean copyOnly) in C:\BuildAgent\work\13f3e374fa512e16\src\NzbDrone.Core\MediaFiles\UpgradeMediaFileService.cs:line 64

   at NzbDrone.Core.MediaFiles.EpisodeImport.ImportApprovedEpisodes.Import(List`1 decisions, Boolean newDownload, DownloadClientItem downloadClientItem, ImportMode importMode) in C:\BuildAgent\work\13f3e374fa512e16\src\NzbDrone.Core\MediaFiles\EpisodeImport\ImportApprovedEpisodes.cs:line 108```

Windows 10. I was not having these issues in prior versions, and it doesn't happen every time.

**Comment by Safihre** [comment]
Are you sure it's SABnzbd and not your virusscanner (Windows Defender) that doesn't trust SABnzbd yet so scans all files it produces?

**Comment by PearsonFlyer** [comment]
I am not using Windows Defender. But, I cannot say for sure it's Sab that's causing it. Plex is also scanning credits, but it didn't appear to be scanning credits on the episode files that it would have been replacing, when I looked at this most recent failure.

I did roll back, as Inpheaux requested, so if it continues to occur, I'll know that sab 4.0.0 didn't cause it and I will update here as appropriate. Thanks!
